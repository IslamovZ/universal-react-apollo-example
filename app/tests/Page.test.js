import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import { print } from 'graphql-tag/printer';

import PageWithData, { PAGE_QUERY, Page } from 'components/Page/Page';

describe('Page Container', () => {
  it('renders without crashing', () => {
    const output = renderer.create(
      <MockedProvider>
        <PageWithData />
      </MockedProvider>,
    );
    expect(output.toJSON()).toMatchSnapshot();
  });
});

describe('Page Query', () => {
  it('should match expected shape', () => {
    expect(print(PAGE_QUERY)).toMatchSnapshot();
  });
});

describe('Page Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Page menu={[{ id: 1, title: 'ok' }]} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
