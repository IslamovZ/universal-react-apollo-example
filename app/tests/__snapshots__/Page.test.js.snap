exports[`Page Component renders correctly 1`] = `
<div>
  <a
    className="toTop"
    onClick={undefined} />
</div>
`;

exports[`Page Container renders without crashing 1`] = `
<div>
  Loading....
</div>
`;

exports[`Page Query should match expected shape 1`] = `
"query getPosts($lang: String!) {
  wp_query {
    posts(category_name: $lang) {
      title
      content
    }
  }
}
"
`;
